core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -306,7 +306,13 @@ object Battle {
         return true
     }
 
-    private fun takeDamage(attacker: ICombatant, defender: ICombatant): Pair<Int, Int> {
+    private class DamageDealt(attackerDealt: Int, defenderDealt: Int)
```

Comment:
I'd just make this is data class, that saves declaring the additional variables:

```kotlin
private data class DamageDealt(val attackerDealt: Int, val defenderDealt: Int) {}
```